### ISSUE TITLE
Allow searching with 1 or 2 characters when JavaScript disabled

### ISSUE THREAD

**Issue by tiansh** [mozilla/addons-frontend#7604]
### Describe the problem and steps to reproduce it:
* Open https://addons.mozilla.org/en-US/firefox/
* Type 2 characters in search box and try to act a searching

### What happened?
* Searching is blocked by front-end, a tool-tip is shown, which ask user to input at least 3 characters.
* Manually changing URL to perform a 2 characters search works fine. So this is not a limitation by backend server. (example: https://addons.mozilla.org/en-US/firefox/search/?q=%E7%99%BE%E5%BA%A6 )
* Update: browser built-in `about:addons` page allow searching with only 1 character.

### What did you expect to happen?
It should allow user to perform a 2 characters search.

### Anything else we should know?
It seems this change is involved by #4532, and @tofumatt asked to config this value as 3 instead of 2.
Most Chinese words are two characters (about 70% ~ 85%), and many Chinese web sites use two characters name. It would be common and reasonable to perform a searching with 2 characters for Chinese language user. (I'm not sure about other languages, but maybe it also applied to some other East-Asian language.) So I would prefer this value to be 2. Or maybe no limitation if it won't break searching backend. Auto completion would be another thing which I don't care whether it require at least 3 or 5 letters.

**Comment by tiansh** [comment]
Update:

I just realize that I can perform a 2 characters search as long as javascript on that page loads correctly. This only happened if javascript failed to load in any case.
Although it perform a correct searching, I would still suggest removing the `minlength` attribute since it still shows tooltip which requires users to type at least 3 characters.

**Comment by willdurand** [comment]
I believe we can indeed set the `minLength` to `2`. Would that work for you @tiansh?

**Comment by tiansh** [comment]
Why not remove the attribute? If JavaScript is enabled, no such validation would performed.

**Comment by willdurand** [comment]
I think there should be a validation in js too.

**Comment by tiansh** [comment]
> 
> 
> I think there should be a validation in js too.

So, maybe 2 should be fine.

**Comment by suhailsinghbains** [comment]
Are we changing the minimum length to 2 instead of 3? @willdurand 
Can I please work on it, if we are doing it?

**Comment by willdurand** [comment]
> Are we changing the minimum length to 2 instead of 3? @willdurand
> Can I please work on it, if we are doing it?

yes and yes.

**Comment by ioanarusiczki** [comment]
@willdurand Search suggestions begin when 2 chars are typed in instead of 3 (as it used to be). 
I filed a minor followup issue for the tooltip - https://github.com/mozilla/addons-frontend/issues/7604

dev versus prod

![dev versus prod](https://user-images.githubusercontent.com/33448286/56959386-ee8b2780-6b55-11e9-93d6-bb9ea52bac85.gif)
